### PR TITLE
Level enhancements

### PR DESCRIPTION
**Pull request by davidhabib**
This pull requests includes:

- validation that a Level name is unique within its ladder
- improved validation when levels saved to tag the correct record with validation errors, such as overlapping another level.
- support for both Minumum Level and Maximum Level being null, meaning no lower/upper limit
- changed treatment of Maximum Level to be exclusive, leaving Min inclusive, ie,  Min <= data < Max.
- additional Level Health Check logic to detect invalid field datatypes
- Level Health Check test code

### CODE REVIEW

**Comment by davidhabib** [comment]
@njjc this is ready for review and merge.  thanks.

File: src/classes/STG_PanelHealthCheck_CTRL.cls [hunk 1]
```diff
@@ -756,12 +756,29 @@ public without sharing class STG_PanelHealthCheck_CTRL extends STG_Panel {
...
                         cErrors++; 
                         createDR(Label.healthLabelVerifyLevels, statusError, 
                             string.format(Label.healthDetailsInvalidLevelField, new string[]{lvl.Name, strField, lvl.Target__c}),
                             string.format(label.healthSolutionInvalidLevelField, new string[]{lvl.Name}));
-                    }        
+                    } 
+                    // ensure the datatype is valid
+                    else if (strField != null) {
+                        Schema.Displaytype dt = UTIL_Describe.getFieldDisplayType(lvl.Target__c, strField);
+                        if (i == 0 && !UTIL_Describe.canDisplaytypesCopy(dt, Schema.DisplayType.Double)) {
```
Comment (njjc) [inline src/classes/STG_PanelHealthCheck_CTRL.cls]: I'm not positive, but if a number field has 0 decimal places, could its [display type](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_enum_Schema_DisplayType.htm) be Integer? Also, couldn't this feature theoretically work fine with percent fields?

Comment (davidhabib) [inline src/classes/STG_PanelHealthCheck_CTRL.cls]: displaytype.double returns true for copying between all of the various numeric types.  so it covers all those cases.

Comment (njjc) [inline src/classes/STG_PanelHealthCheck_CTRL.cls]: Ah gotcha, sounds good. I forgot about that method.

**Pull request merged**